[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: backup started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: backup started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: backup started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: backup started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: backup started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: backup started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: backup started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: Primary
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] maintenance: restore started/running, backup id:'2024-08-18T01:45:20Z'
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: Primary
  config-server/1 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:834 Model status:

Model  Controller          Cloud/Region        Version  SLA          Timestamp
test   microk8s-localhost  microk8s/localhost  3.4.4    unsupported  01:54:57Z

App            Version  Status   Scale  Charm          Channel      Rev  Address         Exposed  Message
application             waiting      1  application                   0  10.152.183.163  no       installing agent
config-server           active       2  mongodb-k8s                   0  10.152.183.218  no       Primary
s3-integrator           active       1  s3-integrator  latest/edge   35  10.152.183.148  no       
shard-one               active       2  mongodb-k8s                   1  10.152.183.243  no       
shard-two               active       1  mongodb-k8s                   2  10.152.183.108  no       Primary

Unit              Workload  Agent  Address     Ports  Message
application/0*    waiting   idle   10.1.50.8          
config-server/0*  active    idle   10.1.50.13         Primary
config-server/1   active    idle   10.1.50.14         
s3-integrator/0*  active    idle   10.1.50.23         
shard-one/0*      active    idle   10.1.50.19         
shard-one/1       active    idle   10.1.50.20         Primary
shard-two/0*      active    idle   10.1.50.24         Primary

[32mINFO    [0m pytest_operator.plugin:plugin.py:840 Juju error logs:

unit-config-server-0: 01:34:17 ERROR unit.config-server/0.juju-log Failed to create the operator user: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 926, in _init_operator_user
    process.wait_output()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 1771, in wait_output
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)
ops.pebble.ExecError: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
unit-config-server-0: 01:34:23 ERROR unit.config-server/0.juju-log Failed to create the operator user: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 926, in _init_operator_user
    process.wait_output()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 1771, in wait_output
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)
ops.pebble.ExecError: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
unit-config-server-0: 01:35:31 ERROR unit.config-server/0.juju-log config-server:7: Failed to get pbm status: non-zero exit code 1 executing ['/usr/bin/pbm', 'status', '-o', 'json'], stdout='{"Error":"get status of pitr: unable check PITR config status: get config: missed config"}\n', stderr=''
unit-config-server-0: 01:35:41 ERROR unit.config-server/0.juju-log config-server:6: Failed to add shard shard-two to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1723944940, 4), 'signature': {'hash': b'\\xe4\\xae\\x1c\\x00\\xadn\\xbc\\xd3]\\x979\\xf3C\\xdc\\x9d\\x08\\xd4\\x84c\\x06', 'keyId': 7404286823872069656}}, 'operationTime': Timestamp(1723944940, 4)}")
unit-config-server-0: 01:35:41 ERROR unit.config-server/0.juju-log config-server:6: shard-two shard does not have the same auth as the config server.
unit-config-server-0: 01:35:44 ERROR unit.config-server/0.juju-log config-server:6: Failed to add shard shard-one to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1723944944, 7), 'signature': {'hash': b'\\x88H\\x9b\\x1fr\\xb1Y\\xf5\\x18\\xf8\\xec\\x84\\xb1+\\x95\\x0e[\\xea\\xb7\\x15', 'keyId': 7404286823872069656}}, 'operationTime': Timestamp(1723944944, 7)}")
unit-config-server-0: 01:35:45 ERROR unit.config-server/0.juju-log config-server:6: shard-one shard does not have the same auth as the config server.
unit-config-server-0: 01:35:45 ERROR unit.config-server/0.juju-log config-server:6: Failed to get pbm status: non-zero exit code 1 executing ['/usr/bin/pbm', 'status', '-o', 'json'], stdout='{"Error":"get status of cluster: connect to `shard-two` [shard-two/shard-two-0.shard-two-endpoints:27017]: ping: connection() error occurred during connection handshake: auth error: unable to authenticate using mechanism \\"SCRAM-SHA-256\\": (AuthenticationFailed) Authentication failed."}\n', stderr=''
unit-config-server-0: 01:35:46 ERROR unit.config-server/0.juju-log config-server:6: Failed to add shard shard-one to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1723944945, 2), 'signature': {'hash': b'\\xcd\\xeb\\xf4+*\\x1fz8p\\xfc\\xd9w\\x83\\xad\\xda\\x9c&t9\\xd9', 'keyId': 7404286823872069656}}, 'operationTime': Timestamp(1723944945, 2)}")
unit-config-server-0: 01:35:46 ERROR unit.config-server/0.juju-log config-server:6: shard-one shard does not have the same auth as the config server.
unit-config-server-0: 01:36:01 ERROR unit.config-server/0.juju-log config-server:6: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1723944960, 4), \'signature\': {\'hash\': b\'`\\x15\\xc9\\x18\\x8c"\\np\\xb6\\xf5\\xbd\\x85+\\xedP\\x16\\xc0W\\x15Z\', \'keyId\': 7404286823872069656}}, \'operationTime\': Timestamp(1723944960, 4)}')
unit-config-server-0: 01:36:01 ERROR unit.config-server/0.juju-log config-server:6: Failed to add shard-one to cluster
unit-config-server-0: 01:36:01 ERROR unit.config-server/0.juju-log config-server:6: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1723944960, 4), \'signature\': {\'hash\': b\'`\\x15\\xc9\\x18\\x8c"\\np\\xb6\\xf5\\xbd\\x85+\\xedP\\x16\\xc0W\\x15Z\', \'keyId\': 7404286823872069656}}, \'operationTime\': Timestamp(1723944960, 4)}')
unit-config-server-0: 01:36:04 ERROR unit.config-server/0.juju-log config-server:6: Failed to get pbm status: non-zero exit code 1 executing ['/usr/bin/pbm', 'status', '-o', 'json'], stdout='{"Error":"get status of cluster: connect to `shard-one` [shard-one/shard-one-0.shard-one-endpoints:27017,shard-one-1.shard-one-endpoints:27017]: ping: connection() error occurred during connection handshake: auth error: unable to authenticate using mechanism \\"SCRAM-SHA-256\\": (AuthenticationFailed) Authentication failed."}\n', stderr=''
unit-config-server-0: 01:36:04 ERROR unit.config-server/0.juju-log config-server:6: Failed to get pbm status: non-zero exit code 1 executing ['/usr/bin/pbm', 'status', '-o', 'json'], stdout='{"Error":"get status of cluster: connect to `shard-one` [shard-one/shard-one-0.shard-one-endpoints:27017,shard-one-1.shard-one-endpoints:27017]: ping: connection() error occurred during connection handshake: auth error: unable to authenticate using mechanism \\"SCRAM-SHA-256\\": (AuthenticationFailed) Authentication failed."}\n', stderr=''
unit-config-server-0: 01:36:05 ERROR unit.config-server/0.juju-log config-server:6: Failed to get pbm status: non-zero exit code 1 executing ['/usr/bin/pbm', 'status', '-o', 'json'], stdout='{"Error":"get status of cluster: connect to `shard-one` [shard-one/shard-one-0.shard-one-endpoints:27017,shard-one-1.shard-one-endpoints:27017]: ping: connection() error occurred during connection handshake: auth error: unable to authenticate using mechanism \\"SCRAM-SHA-256\\": (AuthenticationFailed) Authentication failed."}\n', stderr=''
unit-shard-one-0: 01:40:25 ERROR unit.shard-one/0.juju-log Backup failed: Relation with s3-integrator charm missing, cannot restore from a backup.
unit-shard-one-0: 01:40:27 ERROR unit.shard-one/0.juju-log List-backups failed: Relation with s3-integrator charm missing, cannot restore from a backup.
unit-shard-one-0: 01:40:28 ERROR unit.shard-one/0.juju-log Restore failed: Relation with s3-integrator charm missing, cannot restore from a backup.

[32mINFO    [0m pytest_operator.plugin:plugin.py:904 Forgetting main...